src/java.base/share/classes/jdk/internal/classfile/CodeBuilder.java

```diff
@@ -1192,6 +1234,14 @@ default CodeBuilder lookupswitch(Label defaultTarget, List<SwitchCase> cases) {
         return lookupSwitchInstruction(defaultTarget, cases);
     }
 
+    default CodeBuilder lookupswitch(Consumer<SwitchBuilder> switchHandler) {
+        return block(b -> {
```

Comment:
Maybe we should convert this into an anonymous or local class, for these switch building utilities might be useful when migrating the switching MethodHandle's LambdaForm invoker code.

Reply:
Conversion to anonymous classes should be done only when specifically needed or we will end with horrible codebase.
Code used during JDK bootstrap must be very carefully selected, as for example any use of BlockCodeBuilder involves several other classes and would require to expand also several switch expressions to if/then/else sequences.
I would recommend to use bare minimum (only the base builder features) from the Classfile API to construct lambdas and method handles.